main.go

```diff
@@ -23,12 +23,16 @@ func main() {
 	c := prepareConfig()
 	log.Printf("configuration: %+v", c)
 
-	client, err := connectMqtt(c.mqtt.server)
+	client, err := connectMqtt(c)
```

Comment:
Why are we sending the entire config object to a function that is connecting to mqtt?